sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/http/models/HttpLogOptions.java

```diff
@@ -0,0 +1,312 @@
+     *
+     * @return The list of allowed headers.
+     */
+    public List<HeaderName> getAllowedHeaderNames() {
+        return Collections.unmodifiableList(allowedHeaderNames);
+    }
+
+    /**
+     * Sets the given allowed headers that should be logged.
+     *
+     * <p>
+     * This method sets the provided header names to be the allowed header names which will be logged for all HTTP
+     * requests and responses, overwriting any previously configured headers. Additionally, users can use
+     * {@link HttpLogOptions#addAllowedHeaderName(String)} or {@link HttpLogOptions#getAllowedHeaderNames()} to add or
+     * remove more headers names to the existing set of allowed header names.
+     * </p>
+     *
+     * @param allowedHeaderNames The list of allowed header names from the user.
+     *
+     * @return The updated HttpLogOptions object.
+     */
+    public HttpLogOptions setAllowedHeaderNames(final List<HeaderName> allowedHeaderNames) {
+        this.allowedHeaderNames = allowedHeaderNames == null ? new ArrayList<>() : allowedHeaderNames;
+
+        return this;
+    }
+
+    /**
+     * Sets the given allowed header to the default header set that should be logged.
+     *
+     * @param allowedHeaderName The allowed header name from the user.
+     *
+     * @return The updated HttpLogOptions object.
+     *
+     * @throws NullPointerException If {@code allowedHeaderName} is {@code null}.
+     */
+    public HttpLogOptions addAllowedHeaderName(final String allowedHeaderName) {
+        Objects.requireNonNull(allowedHeaderName);
+        this.allowedHeaderNames.add(HeaderName.fromString(allowedHeaderName));
+
+        return this;
+    }
+
+    /**
+     * Gets the allowed query parameters.
+     *
+     * @return The list of allowed query parameters.
+     */
+    public Set<String> getAllowedQueryParamNames() {
+        return Collections.unmodifiableSet(allowedQueryParamNames);
+    }
+
+    /**
+     * Sets the given allowed query params to be displayed in the logging info.
+     *
+     * @param allowedQueryParamNames The list of allowed query params from the user.
+     *
+     * @return The updated HttpLogOptions object.
+     */
+    public HttpLogOptions setAllowedQueryParamNames(final Set<String> allowedQueryParamNames) {
+        this.allowedQueryParamNames = allowedQueryParamNames == null ? new HashSet<>() : allowedQueryParamNames;
+
+        return this;
+    }
+
+    /**
+     * Sets the given allowed query param that should be logged.
+     *
+     * @param allowedQueryParamName The allowed query param name from the user.
+     *
+     * @return The updated HttpLogOptions object.
+     *
+     * @throws NullPointerException If {@code allowedQueryParamName} is {@code null}.
+     */
+    public HttpLogOptions addAllowedQueryParamName(final String allowedQueryParamName) {
+        this.allowedQueryParamNames.add(allowedQueryParamName);
+        this.getClass().getName();
```

Comment:
Was this added for debugging?

Reply:
Apparently its in main too https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/core/azure-core/src/main/java/com/azure/core/http/policy/HttpLogOptions.java#L179 
:octocat: